sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnSignatures.java

```diff
@@ -505,6 +513,56 @@ private static void verifyUnsplittableMethods(ErrorReporter errors, DoFnSignatur
             formatType(paramT),
             formatType(expectedOutputReceiverT));
         extraParameters.add(DoFnSignature.Parameter.outputReceiver());
+      } else if (Timer.class.equals(rawType)) {
+        // m.getParameters() is not available until Java 8
+        Annotation[] annotations = m.getParameterAnnotations()[i];
+        String id = null;
+        for (Annotation anno : annotations) {
+          if (anno.annotationType().equals(DoFn.TimerId.class)) {
+            id = ((DoFn.TimerId) anno).value();
+            break;
+          }
+        }
+        errors.checkArgument(
+            id != null,
+            "%s parameter of type %s at index %s missing %s annotation",
+            fnClass.getRawType().getName(),
+            params[i],
+            i,
+            DoFn.TimerId.class.getSimpleName());
+
+        errors.checkArgument(
+            !stateParameters.containsKey(id),
+            "%s parameter of type %s at index %s duplicates %s(\"%s\") on other parameter",
+            fnClass.getRawType().getName(),
+            params[i],
+            i,
+            DoFn.TimerId.class.getSimpleName(),
+            id);
+
+        TimerDeclaration timerDecl = timerDeclarations.get(id);
+        errors.checkArgument(
+            timerDecl != null,
+            "%s parameter of type %s at index %s references undeclared StateId \"%s\"",
+            fnClass.getRawType().getName(),
+            params[i],
+            i,
+            id);
+
+        errors.checkArgument(
+            timerDecl.field().getDeclaringClass().equals(m.getDeclaringClass()),
+            "Method %s has State parameter at index %s for state %s"
```

Comment:
Another copy-paste error. Please check for more, or consider finding a way to deduplicate.